src/jit/importer.cpp

```diff
@@ -3097,7 +3097,7 @@ GenTreePtr      Compiler::impIntrinsic(CORINFO_CLASS_HANDLE     clsHnd,
         return op1;
 
 
-#ifndef _TARGET_ARM_
+#ifdef _TARGET_XARCH_
```

Comment:
There is an issue,  genLockedInstructions #2823 
But we probably can not easily implement these x86/x64 instructions on the Arm32 or Arm64 architectures, so they become method calls into the VM rather than instrinsic instructions.